fontforgeexe/cvpalettes.c

```diff
@@ -1558,6 +1559,42 @@ static BDFChar *BDFCharFromLayer(SplineChar *sc,int layer) {
 return( SplineCharAntiAlias(&dummy,ly_fore,24,4));
 }
 
+/**
+ *  \brief Recalculate the number of visible layers and reposition the scrollbar
+ *
+ *  \param [in] cv The charview
+ */
+static void CVLayers2Reflow(CharView *cv) {
+    extern int _GScrollBar_Width;
+    GGadget *scrollbar;
+    GRect cvl2size;
+
+    GDrawGetSize(cvlayers2, &cvl2size);
+    // Minus 2 because we always have the 'Guide' and 'Back' entries
+    layer2.visible_layers = (cvl2size.height - layer2.header_height - 2 * CV_LAYERS2_LINE_HEIGHT) / CV_LAYERS2_LINE_HEIGHT;
+    if (layer2.visible_layers < 0) {
+        layer2.visible_layers = 0;
+    }
+
+    scrollbar = GWidgetGetControl(cvlayers2, CID_SB);
+    // Check if we need the scrollbar or not.
+    if (layer2.current_layers - 2 <= layer2.visible_layers || layer2.visible_layers <= 0) {
```

Comment:
I've modified the check to not even bother showing the scrollbar if there isn't any room to display any modifiable layers (this can happen if the window is undocked and the user resizes to a height smaller than 3 entries high).